targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/TARGET_MCU_NRF52840/TARGET_EP_AGORA/ONBOARD_TELIT_ME910.cpp

```diff
@@ -0,0 +1,140 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2019 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if MBED_CONF_NSAPI_PRESENT
+
+#if EP_AGORA_ENABLE_CELL
```

Comment:
Any way to only include this file when needed in the build versus #ifdef'ing out its contents?

Reply:
@40Grit The reason I put the `#if` in was so that someone can change the `ep-agora.enable-cell` setting in their `mbed_app.json` file and toggle whether the contents of this file gets built. I'm not sure how it effects building/linking exactly though...

Reply:
Understand why it's there.

I'm questioning moreso if there is a way around needing to have the ifdef in the sources via build configuration that just doesn't include the file in the build.

Specifically question is for the tools team. But I'm on my personal account so I cannot @mention them.aa

Reply:
@ARMmbed/mbed-os-tools

Reply:
It's not , we use this also for HAL files `#if DEVICE_I2C` for instance to exclude boards do not have i2c for instance (sharing common MCU family).

Reply:
This looks like a reasonable thing to want, but it looks like the build system doesn't support the generic inclusion or exclusion of a file for compilation based on a configuration parameter. It's something we'll consider as we develop the build system in future. For now the #if defined solution looks like the way to do this.